backend/helpers/csv_format_checker.py

```diff
@@ -0,0 +1,83 @@
+from .csv_column_headers import get_csv_headers, csv_column_headers
+import pandas as pd
+from .csv_exceptions import MissingHeadersException, ExtraHeadersException, InvalidISBN13LengthException, NotInDbException, InsufficientStockException, NotAnIntegerException, NegativeValueException, NotANumberException
+from typing import Callable
+from books.models import Book
+
+
+class CSVFormatChecker:
+
+    def __init__(self, csv_import_type: str) -> None:
+        self.expected_headers = get_csv_headers(csv_import_type)
+        self.price = csv_column_headers.get(csv_import_type)[0]
+
+    def are_headers_correct(self, actual_headers: list) -> None:
+        actual_headers = [header.lower() for header in actual_headers]
+        missing_headers = [header for header in self.expected_headers if header not in actual_headers]
+        if len(missing_headers) != 0:
+            raise MissingHeadersException(missing_headers)
+        extra_headers = [header for header in actual_headers if header not in self.expected_headers]
+        if len(extra_headers) != 0:
+            raise ExtraHeadersException(extra_headers)
+
+    def find_malformed_data(self, csv_df: pd.DataFrame) -> list:
+        """Returns for each row the columns with malformed data"""
+        csv_df = csv_df.rename(columns={"isbn_13": "isbn13"})
+        malformed_cells = [{} for _ in range(len(csv_df.index))]
+
+        for index, row in csv_df.iterrows():
+            for column in csv_df.columns:
+                try:
+                    getattr(self, f"{column.rsplit('_', 1)[-1]}_conditions")(row)
```

Comment:
Maybe it's because I don't use much python, but this "getattr" line isn't super intuitive for me, especially for such an important part of the code. Maybe a comment or split it up into multiple lines?

Reply:
Added comment. And this isn't just you being inexperienced in Python. This is a pretty complex piece of code